src/file/properties.rs

```diff
@@ -0,0 +1,550 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Reader and writer properties.
+//!
+//! # Usage
+//!
+//! ```rust
+//! use parquet::basic::{Compression, Encoding};
+//! use parquet::file::properties::*;
+//! use parquet::schema::types::ColumnPath;
+//!
+//! // Create properties with default configuration.
+//! let props = WriterProperties::builder().build();
+//!
+//! // Use properties builder to set certain options and assemble the configuration.
+//! let props = WriterProperties::builder()
+//!   .set_writer_version(WriterVersion::PARQUET_1_0)
+//!   .set_encoding(Encoding::PLAIN)
+//!   .set_column_encoding(ColumnPath::from("col1"), Encoding::DELTA_BINARY_PACKED)
+//!   .set_compression(Compression::SNAPPY)
+//!   .build();
+//!
+//! assert_eq!(props.writer_version(), WriterVersion::PARQUET_1_0);
+//! assert_eq!(props.encoding(&ColumnPath::from("col1")), Encoding::DELTA_BINARY_PACKED);
+//! assert_eq!(props.encoding(&ColumnPath::from("col2")), Encoding::PLAIN);
+//! ```
+
+use std::collections::HashMap;
+use std::rc::Rc;
+
+use basic::{Compression, Encoding};
+use schema::types::ColumnPath;
+
+// TODO: Add reader properties.
+
+const DEFAULT_PAGE_SIZE: usize = 1024 * 1024;
+const DEFAULT_WRITE_BATCH_SIZE: usize = 1024;
+const DEFAULT_WRITER_VERSION: WriterVersion = WriterVersion::PARQUET_1_0;
+const DEFAULT_ENCODING: Encoding = Encoding::PLAIN;
+const DEFAULT_COMPRESSION: Compression = Compression::UNCOMPRESSED;
+const DEFAULT_DICTIONARY_ENABLED: bool = true;
+const DEFAULT_DICTIONARY_PAGE_SIZE_LIMIT: usize = DEFAULT_PAGE_SIZE;
+const DEFAULT_STATISTICS_ENABLED: bool = true;
+const DEFAULT_MAX_STATISTICS_SIZE: usize = 4096;
+const DEFAULT_MAX_ROW_GROUP_LENGTH: usize = 64 * 1024 * 1024;
+const DEFAULT_CREATED_BY: &str = env!("PARQUET_CREATED_BY");
```

Comment:
I rely on the fact that this will be substituted on actual `&str` when compiling, so that it will be part of the binary.